### PR TITLE
Mul imul matmul

### PR DESCRIPTION
**Pull request by AGaliciaMartinez**
This adds mul, neg, imul and matmul.

Benchmark for matmul:

matmul_dense

### CODE REVIEW

File: src/qutip_tensorflow/core/data/__init__.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
 from .convert import *
 from .tftensor import *
 from .add import *
+from .matmul import *
+from .mul import *
+
```
Comment (AGaliciaMartinez) [inline src/qutip_tensorflow/core/data/__init__.py:6]:
```suggestion
```

**Comment by AGaliciaMartinez** [comment]
Is it possible to see somehow which lines are missing coverage? When I got to coverall report it says "Source Not Available".

File: src/qutip_tensorflow/core/data/matmul.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            + str((left.shape[0], right.shape[1]))
+        )
+
+
+def matmul_tftensor(left, right, scale=1, out=None):
+    """
+    Perform the operation
+        ``out := scale * (left @ right) + out``
+    where `left`, `right` and `out` are matrices.  `scale` is a complex scalar,
+    defaulting to 1. If `out` not gives is assumed to be 0.
```
Comment (hodgestar) [inline src/qutip_tensorflow/core/data/matmul.py:37]:
```suggestion
    defaulting to 1. If `out` is not given it is assumed to be 0.
```

**Review by hodgestar — APPROVED** [review]
Left one docstring suggestion, but looks good to me.

**Review by Ericgig — APPROVED** [review]

**Pull request merged**